### ISSUE TITLE
Mach var restricted by airspeed var

### ISSUE THREAD

**Issue by rthom91** [flybywiresim/aircraft#8294]
### Aircraft Version

Stable

### Build info

```json
{
    "built": "2023-11-03T17:47:36+00:00",
    "ref": "refs/tags/v0.11.1",
    "sha": "f055eb02dc0af32441f8b2873f58e8316d7ca638",
    "actor": "Saschl",
    "event_name": "manual",
    "pretty_release_name": "stable/v0.11.1",
    "version": "v0.11.1-rel.f055eb0"
}
```

### Describe the bug

AUTOPILOT MACH HOLD VAR is being held back by AUTOPILOT AIRSPEED HOLD VAR. MACH does not drop below 0.15 (100 knots) or higher than 0.61 (399 knots).

### Expected behavior

MACH simvar should go to 0.10 min and 0.99 max.

### Steps to reproduce

https://www.youtube.com/watch?v=SxHnrRToAeA

**Comment by Maximilian-Reuter** [comment]
> It looks like the issue has already been solved in the dev version. ![Microsoft Flight Simulator Screenshot 2023 12 07 - 15 55 44 85](https://private-user-images.githubusercontent.com/64700740/288797375-68ed10cc-5003-4ce0-a1d7-53261d94f370.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5NjE0MzQsIm5iZiI6MTcwMTk2MTEzNCwicGF0aCI6Ii82NDcwMDc0MC8yODg3OTczNzUtNjhlZDEwY2MtNTAwMy00Y2UwLWExZDctNTMyNjFkOTRmMzcwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA3VDE0NTg1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFiNGFmOTY5MDU2M2JjMzkzMGI4Y2Q0YTIwOGRjZDY0OGE4YjJhM2VhOGU3NGM3MDc1NjAzMGE3NjNhZWZlMTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.74w-OHdDG9rn_lv_VD6yGn8WY53wT8xgJV-jelfg1G8)

Did you check the Simvars in question ? seeing as the issue wasnt what the FCU shows but what is written to the Simvar which can be seen in the tooltips

**Comment by ArtemisPlayer** [comment]
Mea culpa, it looks like the mach value is converted to speed before being given to the AP.

**Comment by ArtemisPlayer** [comment]
Hey ! 
I am trying to solve the issue, at first it looks simple: I can change this part of the code (fbw-a32nx\src\base\flybywire-aircraft-a320-neo\html_ui\Pages\VCockpit\Instruments\Airliners\FlyByWire_A320_Neo\FCU\A320_Neo_FCU.js):
```JS
            // get current target speed
            let targetSpeed = (isMachActive || this.selectedValue < 1)
                ? SimVar.GetGameVarValue("FROM MACH TO KIAS", "number", this.selectedValue)
                : this.selectedValue;
            // clamp speed into valid range
            targetSpeed = this.clampSpeed(targetSpeed);
            // set target speed
            if (targetSpeed !== this.targetSpeed) {
                Coherent.call("AP_SPD_VAR_SET", 0, targetSpeed).catch(console.error);
                this.targetSpeed = targetSpeed;
            }
            // detect mismatch
            if (Simplane.getAutoPilotAirspeedHoldValue() !== this.targetSpeed) {
                Coherent.call("AP_SPD_VAR_SET", 0, targetSpeed).catch(console.error);
            }
```
into 
```JS
            if (isMachActive || this.selectedValue < 1){
                targetMach = this.clampMach(this.selectedValue); // clamp into valid range
                if (targetMach !== this.targetMach) {
                    // set target mach
                    Coherent.call("AP_MACH_VAR_SET", 0, Math.round(targetMach * 100).catch(console.error));
                    this.targetMach = targetMach;
                }
                if (Simplane.getAutoPilotMachHoldValue() !== this.targetMach) {
                    Coherent.call("AP_MACH_VAR_SET", 0, Math.round(targetMach * 100).catch(console.error));
                }
            } else {
                targetSpeed = this.clampSpeed(this.selectedValue); // clamp into valid range
                if (targetSpeed !== this.targetSpeed) {
                    // set target speed
                    Coherent.call("AP_SPD_VAR_SET", 0, targetSpeed).catch(console.error);
                    this.targetSpeed = targetSpeed;
                }
                // detect mismatch
                if (Simplane.getAutoPilotAirspeedHoldValue() !== this.targetSpeed) {
                    Coherent.call("AP_SPD_VAR_SET", 0, targetSpeed).catch(console.error);
                }
            }
```
(the mach is multiplied by 100 as indicated here : [https://docs.flightsimulator.com/html/Programming_Tools/Event_IDs/Aircraft_Autopilot_Flight_Assist_Events.htm](https://docs.flightsimulator.com/html/Programming_Tools/Event_IDs/Aircraft_Autopilot_Flight_Assist_Events.htm))

But that makes the module crash (only dashed lines).
I have some beginner's questions: 
+ I see no error in the debug console, should I look elsewhere for error messages ? (Does `console.log` logs into the debug console ?)
+ If there is a compilation issue with my code, would I be informed during compilation ?
+ What's the second argument of `Coherent.call` ? 

Thanks for any help !

**Comment by lukecologne** [comment]
> Mea culpa, it looks like the mach value is converted to speed before being given to the AP. Edit: I am working on it

Correct, the AP currently only uses `AUTOPILOT AIRSPEED HOLD VAR`, which is set by the FCU converted from the selected mach.
Anyways, the current legacy FCU is planned to be replaced by a totally new implementation with #7587, which will then also no longer use the sim's default AP variables.

> EDIT: the idea is to not convert mach into speed before sending it to the autopilot, so it does clamp the value to knots limits.

This will not work, we use our own AP which, as I said, currently just reads the sim's default variables. It expects to receive the speed target in CAS. (Note, this is not correct IRL, where the FCU will send selected mach and CAS as A429 words, but that is not currently implemented, and will be corrected in #7587.)

**Comment by Maximilian-Reuter** [comment]
> > Mea culpa, it looks like the mach value is converted to speed before being given to the AP. Edit: I am working on it
> 
> Correct, the AP currently only uses `AUTOPILOT AIRSPEED HOLD VAR`, which is set by the FCU converted from the selected mach. Anyways, the current legacy FCU is planned to be replaced by a totally new implementation with #7587, which will then also no longer use the sim's default AP variables.
> 
> > EDIT: the idea is to not convert mach into speed before sending it to the autopilot, so it does clamp the value to knots limits.
> 
> This will not work, we use our own AP which, as I said, currently just reads the sim's default variables. It expects to receive the speed target in CAS. (Note, this is not correct IRL, where the FCU will send selected mach and CAS seperately as A429 words, but that is not currently implemented, and will be corrected in #7587.)

So that effectively means, that these simvars will be useless once that implementation comes into effect ? If so then this is a "Not planned" then right ?

**Comment by tracernz** [comment]
> Yep, they won't be used anymore and possible not even exist (since I might completely disable the sim AP in the configs, which might result in these not being defined by the sim anymore).

We should look into using the managed mode so we can override the vars, for the benefit of third party tools.
